### PR TITLE
Deployment unit tests

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Initial PR to increase test coverage for machineDeployment controller

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/machineset/controller.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ var stateConfirmationInterval = 100 * time.Millisecond
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
 	r := newReconciler(mgr)
-	return add(mgr, r, r.MachineSetToMachines)
+	return add(mgr, r, r.MachineToMachineSets)
```
Comment (enxebre) [inline pkg/controller/machineset/controller.go]: analogous to https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machinedeployment/controller.go#L258

File: pkg/controller/machinedeployment/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+						MatchLabels: map[string]string{
+							"foo": "bar",
+						},
+					},
+				},
+			},
+		},
+	}
+	controller := true
+	ms := v1alpha1.MachineSet{
```
Comment (frobware) [inline pkg/controller/machinedeployment/controller_test.go]: Given that we enumerate the other instances (ms2, ms3), why not make this `ms1`?

File: pkg/controller/machinedeployment/controller_test.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	}
+	machineDeplopymentList := &v1alpha1.MachineDeploymentList{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "MachineDeploymentList",
+		},
+		Items: []v1alpha1.MachineDeployment{
+			machineDeployment,
+		},
+	}
+	ms := v1alpha1.MachineSet{
```
Comment (frobware) [inline pkg/controller/machinedeployment/controller_test.go]: Given that we enumerate the others, why not make this `ms1`?

File: pkg/controller/machinedeployment/controller.go [hunk 4]
```diff
@@ -194,7 +205,18 @@ func (r *ReconcileMachineDeployment) getMachineDeploymentsForMachineSet(ms *v1al
...
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/168
+		Raw: &metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: v1alpha1.SchemeGroupVersion.String(),
+				Kind:       "MachineDeployment",
+			},
+		},
+	}
+	err := r.Client.List(context.Background(), listOptions, dList)
 	if err != nil {
```
Comment (vincepri) [inline pkg/controller/machinedeployment/controller.go]: Same here, can be simplified with `if err := ...; err != nil {`.

File: pkg/controller/machinedeployment/controller.go [hunk 5]
```diff
@@ -230,7 +252,7 @@ func (r *ReconcileMachineDeployment) getMachineMapForDeployment(d *v1alpha1.Mach
 		return nil, err
 	}
 	machines := &v1alpha1.MachineList{}
-	err = r.List(context.Background(), client.InNamespace(d.Namespace).MatchingLabels(selector), machines)
+	err = r.Client.List(context.Background(), client.InNamespace(d.Namespace).MatchingLabels(selector), machines)
 	if err != nil {
```
Comment (vincepri) [inline pkg/controller/machinedeployment/controller.go]: Should we use `client.ListOptions` here? Mostly to be consistent.

**Comment by roberthbailey** [comment]
/approve
/hold

Please remove the hold once all comments have been addressed.

**Comment by enxebre** [comment]
thanks all for having a look, all comments addressed now

**Comment by chaosaffe** [comment]
/lgtm
Reiterate [lgtm](https://github.com/kubernetes-sigs/cluster-api/pull/646#pullrequestreview-187092880) as all changes have been made as requested

**Pull request merged**